FUSION_Main.py

```diff
@@ -4005,17 +4005,17 @@ def upload_data(self,wsi_file,omics_file,wsi_file_flag,omics_file_flag):
             disable_upload_type = True
             
             if 'Omics' in self.upload_check:
-                return slide_qc_results, thumb_fig, [wsi_upload_children], [omics_upload_children], structure_type_disabled, post_upload_style, disable_upload_type, json.dumps({'plugin_used': {'upload': { 'type': 'Visium' } } })
+                return slide_qc_results, thumb_fig, [wsi_upload_children], [omics_upload_children], structure_type_disabled, post_upload_style, disable_upload_type, json.dumps({'plugin_used': 'upload', 'type': 'Visium' })
             else:
-                return slide_qc_results, thumb_fig, [wsi_upload_children], [], structure_type_disabled, post_upload_style, disable_upload_type, json.dumps({'plugin_used': {'upload': { 'type': 'non-Omnics' } } })
+                return slide_qc_results, thumb_fig, [wsi_upload_children], [], structure_type_disabled, post_upload_style, disable_upload_type, json.dumps({'plugin_used': 'upload', 'type': 'non-Omnics' })
         else:
 
             disable_upload_type = True
 
             if 'Omics' in self.upload_check:
-                return no_update, no_update,[wsi_upload_children], [omics_upload_children], True, no_update, disable_upload_type, json.dumps({'plugin_used': {'upload': { 'type': 'Visium' } } })
+                return no_update, no_update,[wsi_upload_children], [omics_upload_children], True, no_update, disable_upload_type, json.dumps({'plugin_used': 'upload', 'type': 'Visium' })
```

Comment:
Switch this to no_update. This part is triggered if one file is uploaded but it's still waiting on another so it would just record twice.